enterprise/backend/src/metabase_enterprise/workspaces/merge.clj

```diff
@@ -3,22 +3,36 @@
    All functions assumes that validation of the dependencies AFTER merge is done BEFORE calling any of these methods."
   (:require
    [metabase-enterprise.transforms.api :as transforms.api]
+   [metabase-enterprise.workspaces.models.workspace-merge]
+   [metabase-enterprise.workspaces.models.workspace-merge-transform]
    [metabase-enterprise.workspaces.types :as ws.types]
    [metabase.util.malli :as mu]
    [toucan2.core :as t2])
   (:import (java.sql Connection)))
 
 (set! *warn-on-reflection* true)
 
+(defn- create-merge-transform-history!
+  "Create a workspace_merge_transform record to track this merge operation.
+   Does nothing for :noop operations (archived transforms that don't exist globally)."
+  [op merge-map]
+  (when (not= op :noop)
+    (t2/insert! :model/WorkspaceMergeTransform merge-map)))
```

Comment:
I wonder what we should do when we delete a transform.

On the one hand, it's very interesting to have this in the Workspace history.

On the other hand, our only use case right now is individual transform history, so we want an FK, and don't need the history anymore if the transform is deleted (CASCADE).

I lean towards not creating joins on delete right now, and having that CASCADE fk.

We can revisit when we add other ways of viewing the history where deletions are relevant.